### DISCUSSION TITLE
Giex QT06 Smart valve - different Manufacturer / Not working...

### DISCUSSION THREAD

**Discussion by Batou38** [Koenkk/zigbee2mqtt#13487] [Devices]
Hello,
I just received few Smart valves GiEX 
they are looking exactly like that one [https://www.zigbee2mqtt.io/devices/QT06.html](url)

They are detected by Z2M (runing version 1.27.0 on Home assistant) 
but they are shown as a device not managed.

I am not sure whether this is the reason or not but my devices are announcing themselves with Model TS0601 and manufacturer  : _TZE200_a7sghmms....

This seems different from the Zigbee vendor that is defiiend in the zigbee converter : 
TS0601 and Manufacturer : _TZE200_sh1btabb

I am stuck there (I am not an expert), any idea on how I could make that working ?
Thank you.

**Comment by Batou38** [comment]
Ok 
I got it working trough copy pasting the external converter Giex.js just changing the manufacturer code... 
I don't understand what I did wrong in my first attemp but now its working...

**Reply by hkoci** [reply]
Thank you @Batou38, I can confirm this works on my sprinkler by changing the manufacturer code from giex.js

**Comment by hkoci** [comment]
I recently purchased a Zigbee garden valve as well and has the same manufacturer string, _TZE200_a7sghmms.

I was able to get this working on Zigbee2Mqtt by downloading the [giex.js](https://github.com/Koenkk/zigbee-herdsman-converters/blob/21a66c05aa533de356a51c8417073f28092c6e9d/devices/giex.js) file and changing the imports and manufacturer code.

In case anyone else has purchased this a7sghmms model, you'll need to manually import the giex.js code for the sh1btabb and then modify the manufacturer to a7sghmms.

- Changed the imports at the top of imported repo js to include zigbee-herdsman-converters as this is taken directly from the repo source.

```
const exposes = require('zigbee-herdsman-converters/lib/exposes'); 
const tuya = require('zigbee-herdsman-converters/lib/tuya');
```

- Modify manufacturer code to support the new product (the functionality seems to all be working, just a change of manufacturer it looks like).

`fingerprint: [{modelID: 'TS0601', manufacturerName: '_TZE200_a7sghmms'}],`

- Insert this js file into the zigbee2mqtt folder (if running HA, this is the /config/zigbee2mqtt folder)
- Add the giex.js external adapter to the zigbee2mqtt configuration (in HA, this is /config/zigbee2mqtt/configuration.yaml)
```
external_converters:
  - giex.js
```

**Reply by francescros** [reply]
Hi! Thanks, it does work for me.
But the units are not meeting reality: where says 'minutes' it turns to be 'seconds'. Anyw way to overwrite/modify the original importing code?
I'm managing repetitions by Home Assistant, so all automation in HA and sending 'turn on' when wanted, but kept the 'time on' configured in the valve.

**Reply by danruck** [reply]
I've got the same one.  It's in seconds but says minutes for me too. Took me a bit to figure out why it kept stopping after 5 seconds when I was testing it.  Easy to work around once you know I guess.

**Comment by iznaf** [comment]
Hi guys, 
My valve is also not supported…
How can I check the manufacturer name? In HB frontend it doesn’t show… Only the device id.

**Comment by mmirannalti** [comment]
Thanks everyone, this post saved me from quite an headache!
